app.ts

```diff
@@ -0,0 +1,230 @@
+	let firstNumber, lastNumber;
+	if (myClass.firstNumber < 0 || myClass.firstNumber > recordCount) {
+		firstNumber = 0;
+	} else {
+		firstNumber = myClass.firstNumber;
+	}
+	lastNumber = firstNumber + calculatedRows - 1;
+	if (lastNumber >= recordCount) {
+		lastNumber = recordCount;
+		firstNumber = lastNumber - (calculatedRows - 1);
+	}
+	return { firstNumber, lastNumber };
+}
+
+function updateArrowVisibility(firstNumber: number, lastNumber: number, recordCount: number) {
+	if (firstNumber === 0) {
+		$('.arrow-left').hide();
+	} else {
+		$('.arrow-left').show();
+	}
+
+	if (lastNumber >= recordCount) {
+		$('.arrow-right').hide();
+	} else {
+		$('.arrow-right').show();
+	}
+}
+
+function fetchAndDisplayRecords(firstNumber: number, lastNumber: number, inputValue: number) {
+	return myClass.fetchRecords(firstNumber, lastNumber)
+		.then(records => {
+			const tbody = $("tbody");
+			tbody.empty();
+			for (const record of records) {
+				/** creates row for each record*/
+				$("tbody").append(`<tr class="row"></tr>`);
+				const lastRow = $(".row:last");
+				for (const value of record) {
+					/** assign each record to their column in a specified row */
+					lastRow.append(`<td>${value}</td>`);
+				}
+				if (record.includes(inputValue)) {
+					/** highlights the searched row */
+					lastRow.css('background-color', '#DDC0B4');
+				}
+				$("tbody").append(lastRow);
+			}
+			return records;
+		})
+		.catch(err => {
+			throw new Error('Error fetching records' + err);
+		});
+}
+
+/** recalculates the record range that includes inputValue */
+function searchRecordsAndResize(recordCount: number) {
+	let inputValue = $('#searchInput').val() as number;
+	if (inputValue < 0 || inputValue > recordCount) {
+		$('.modal').css('display', 'block');
+		$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+		$('#page').empty().append(`Showing record: ${myClass.firstNumber} - ${myClass.lastNumber}`);
+		$('#searchInput').val('');
+		return;
+	}
+	let calculatedRows = calculatingRows();
+	/** divides the calculated max rows in half*/
+	const halfRange = Math.floor(calculatedRows / 2);
+	myClass.firstNumber = Math.max(0, inputValue - halfRange);
+	myClass.lastNumber = Math.min(recordCount, myClass.firstNumber + (calculatedRows - 1));
+	displayRecords(recordCount);
+}
+
+/** Navigates to the next set of records */
+function rightArrow(recordCount: number) {
+	$('#page').empty();
+	$('#searchInput').val('');
+	/** retrieves the last row */
+	const lastRow = document.querySelector("#recordsTable tbody .row:last-child");
+	/** checks if the last row exists */
+	if (lastRow) {
+		const cells = lastRow.querySelectorAll("td");
+		const lastRecord = [];
+		for (const cell of Array.from(cells)) {
+			lastRecord.push(cell.textContent || "");
+		}
+		/** determines te value in the last row */
+		const lastID = parseFloat(lastRecord[0]);
```

Comment:
Instead of looping over everything... can't you rather just check the length of `cells` and then only get the value of the first cell, instead of copying over the value of every cell? Feels like overkill to create a whole new array and populate it if you only want one value.